docs/proposals/gateway-settings.md

```diff
@@ -0,0 +1,176 @@
+
+These settings apply to the `main`, `http`, and/or `stream` contexts of the NGINX config.
+
+To begin, the Gateway Settings config will include the following NGINX directives (focusing on OpenTelemetry tracing):
+
+- [`otel_exporter`](https://nginx.org/en/docs/ngx_otel_module.html#otel_exporter)
+- [`otel_service_name`](https://nginx.org/en/docs/ngx_otel_module.html#otel_service_name)
+- [`otel_span_attr`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_attr): set global span attributes that will be merged with the span attributes set in the [Observability extension](nginx-extensions.md#observability).
+
+In the future, this config will be extended to support other directives, such as those defined in the [NGINX Extensions Proposal](nginx-extensions.md#gateway-settings).
+
+## API, Customer Driven Interfaces, and User Experience
+
+The `GatewaySettings` API is a CRD that is a part of the `gateway.nginx.org` Group. It will be referenced in the `parametersRef` field of a GatewayClass. It will live at the cluster scope.
+
+This is a dynamic configuration that can be changed by a user at any time, and NGF will propagate those changes to NGINX. This is something we need to clearly document in our public documentation about this feature, so that users know that all Gateways under the Class can be updated by these settings.
+
+For example, a `GatewaySettings` named `gw-settings` would be referenced as follows:
+
+```yaml
+kind: GatewayClass
+metadata:
+    name: nginx
+spec:
+    controllerName: gateway.nginx.org/nginx-gateway-controller
+    parametersRef:
+        group: gateway.nginx.org/v1alpha1
+        kind: GatewaySettings
+        name: gw-settings
+```
+
+Below is the Golang API for the `GatewaySettings` API:
+
+### Go
+
+```go
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type GatewaySettings struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the GatewaySettings.
+	Spec GatewaySettingsSpec `json:"spec"`
+
+	// Status defines the state of the GatewaySettings.
+	Status GatewaySettingsStatus `json:"status,omitempty"`
+}
+
+type GatewaySettingsSpec struct {
+	// OtelExporter specifies OpenTelemetry export parameters.
+	// +optional
+	OtelExporter *OtelExporter `json:"otelExporter,omitempty"`
+
+	// OtelServiceName is the "service.name" attribute of the Otel resourc.
+	// Default is 'nginx-gateway-fabric:<gateway-name>'.
+	// +optional
+	OtelServiceName *string `json:"otelServiceName,omitempty"`
+
+	// OtelSpanAttributes are custom key/value attributes that are added to each span.
+	// Variables can be included in the values.
+	// +optional
+	OtelSpanAttributes map[string]string `json:"otelSpanAttributes,omitempty"`
+}
+
+type GatewaySettingsStatus struct {
```

Comment:
Should we consider the case when different controllers use the same resource?

Reply:
Good question, how should we handle that? In theory, you could link the same GatewaySettings to multiple GatewayClasses.

I don't know if we should (or can) prevent that. There could be a list of conditions, one entry for each controller (if there's even a way to link a controller name to a condition). Or, we just don't have conditions on this resource.

Reply:
what about having ControllerName field like here https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1alpha2.PolicyAncestorStatus ?

Reply:
We don't necessarily need conditions. Reference Grants are a good example of a resource without conditions. 

If we can perform all of the validation with OpenAPI schema + CEL I don't think an Accepted condition adds much value. Especially if we add a ResolvedRefs condition to the GatewayClass. The ResolvedRefs condition can indicate if there's a problem with the GatewaySettings.

Reply:
Ah, that could be a way. Didn't realize they already considered this case

Reply:
shall we validate it in NGF too, for the chance of those values not being validated properly but bug, misconfiguration, etc.

Reply:
What's our stance on duplicating CRD validation? 

If we do additional validation in the code, we can report that error in the ResolvedRefs condition on the GatewayClass. Something like `ResolvedRefs/False/InvalidGatewaySettings`

Reply:
general recommendation is here https://gateway-api.sigs.k8s.io/geps/gep-922/#implementers

Reply:
I'm okay with not having Conditions on the object. @kate-osborn you're right there may not be much value in just an Accepted condition, and having both ResolvedRefs and GatewaySettings condition may be redundant.